src/code42cli/click_ext/types.py

```diff
@@ -3,9 +3,27 @@
 from datetime import timedelta
 from datetime import timezone
 
+import chardet
 import click
 from click.exceptions import BadParameter
 
+from code42cli.logger import CliLogger
+
+
+class AutoDecodedFile(click.File):
+    """Attempts to autodetect file's encoding prior to normal click.File processing."""
+
+    def convert(self, value, param, ctx):
+        try:
+            with open(value, "rb") as file:
+                self.encoding = chardet.detect(file.read())["encoding"]
+            if self.encoding is None:
+                CliLogger().log_error(f"Failed to detect encoding of file: {value}")
+        except Exception:
```

Comment:
Is this try / except block needed

Reply:
If the filename doesn't exist the `open()` will raise an exception.

Reply:
Shouldn't we `raise` instead of `pass` if we intend `click.File` to do the exception handling.... by `pass` aren't we suppressing the exception!?

Reply:
Would it be nice to add a (file not exists) test case to confirm if the exception cases works as expected!

Reply:
click.File does its own `open()` on the filename. We're only opening here to do the encoding detection, not actually using the result. 

The "doesn't exist" exception test case is a good call.